src/client/src/.tests/components/chatpage.test.js

```diff
@@ -0,0 +1,164 @@
+	})
+	it('Should have no selected room on mount', () => {
+		expect(suite.Component.state(SELECTED_ROOM)).toBe(EMPTY_MAGIC_STRING)
+	})
+	it('Should be disconnected at startup', () => {
+		expect(suite.Component.state(CONNECTED)).toBe(FALSE)
+	})
+	it('Should have no input at startup', () => {
+		expect(suite.Component.state(INPUT)).toBe(EMPTY_MAGIC_STRING)
+	})
+	it('Should have no messages at startup', () => {
+		expect(suite.Component.state(MESSAGES).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have no userRooms at startup', () => {
+		expect(suite.Component.state(USERROOMS).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should have a username at startup', () => {
+		expect(suite.Component.state(USERNAME)).toBe(DUMMY_USER)
+	})
+	it('Should update joined rooms', () => {
+		const fakeData = {roomId: DUMMY_ROOM, usernames: DUMMY_USER}
+		suite.Component.instance().updateJoinedRooms(fakeData)
+		expect(suite.Component.state(USERROOMS)[0].roomId).toBe(fakeData.roomId)
+		expect(suite.Component.state(USERROOMS)[0].usernames).toBe(fakeData.usernames)
+	})
+	it('Should store a message to sessionStorage', () => {
+		const roomId = DUMMY_ROOM
+		const fakeData = {username: DUMMY_USER, message: DUMMY_MESSAGE, timestamp: new Date().getTime()}
+		suite.Component.instance().storeMessage(roomId, fakeData)
+		const retrievedObject = JSON.parse(window.sessionStorage.getItem(roomId))
+		expect(retrievedObject[ZERO_INDEX].message).toBe(fakeData.message)
+		expect(retrievedObject[ZERO_INDEX].username).toBe(fakeData.username)
+		expect(retrievedObject[ZERO_INDEX].timestamp).toBe(fakeData.timestamp)
+	})
+	it('Should not update messages to state if the selected room is not the same', () => {
+		const roomId = DUMMY_ROOM
+		const fakeData = {username: DUMMY_USER, message: DUMMY_MESSAGE, timestamp: new Date().getTime()}
+		suite.Component.instance().updateMessageState({roomId, ...fakeData})
+		expect(suite.Component.state(MESSAGES).length).toBe(EXPECT_NOTHING)
+	})
+	it('Should update messages to state if the selectedroom is the same as the roomId of the message', () => {
+		const roomId = DUMMY_ROOM
+		const fakeData = {username: DUMMY_USER, message: DUMMY_MESSAGE, timestamp: new Date().getTime()}
+		suite.Component.instance().changeSelectedRoom({roomId})
+		suite.Component.instance().updateMessageState({roomId, ...fakeData})
+		expect(suite.Component.state(MESSAGES).length).toBe(EXPECTED_ELEMENTS_COUNT)
+	})
+	it('Should return no users of room', () => {
+		expect(suite.Component.instance().findUsersOfRoom(DUMMY_ROOM)).toBe(EMPTY_MAGIC_STRING)
+	})
+	it('Should return a user in the room', () => {
+		suite.Component.instance().updateJoinedRooms({usernames: DUMMY_USER, roomId: DUMMY_ROOM})
+		expect(suite.Component.instance().findUsersOfRoom(DUMMY_ROOM)).toBe(DUMMY_USER)
+	})
+	it('Should generate a warning to select a room first', () => {
+		expect(typeof suite.Component.instance().generateMessages())
+			.toBe(typeof {})
```

Comment:
How does it test the `Should generate a warning to select a room first`

Reply:
Will fix